eng/Versions.props

```diff
@@ -26,21 +26,21 @@
   </PropertyGroup>
   <!-- Production Dependencies -->
   <PropertyGroup>
-    <SystemCollectionsImmutableVersion>7.0.0</SystemCollectionsImmutableVersion>
-    <SystemConfigurationConfigurationManagerVersion>7.0.0</SystemConfigurationConfigurationManagerVersion>
+    <SystemCollectionsImmutableVersion>8.0.0-preview.7.23375.6</SystemCollectionsImmutableVersion>
```

Comment:
The SCI version in Visual Studio is controlled by the Roslyn team and it generally takes only stable versions, with nontrivial time to update. We have to match the central VS one by policy. So we can make this move only for our net8.0 target right now.